Kernel/FileSystem/Ext2FS/FileSystem.cpp

```diff
@@ -711,9 +711,9 @@ void Ext2FS::flush_block_group_descriptor_table()
     MutexLocker locker(m_lock);
     auto blocks_to_write = ceil_div(m_block_group_count * sizeof(ext2_group_desc), logical_block_size());
     auto first_block_of_bgdt = first_block_of_block_group_descriptors();
-    auto buffer = UserOrKernelBuffer::for_kernel_buffer((u8*)block_group_descriptors());
+    auto span = UserOrKernelBuffer::for_kernel_buffer((u8*)block_group_descriptors(), sizeof(ext2_group_desc));
```

Comment:
Are you sure the size is correct? Shouldn't the size be dependent on `blocks_to_write`?
Also, why are you calling it a `span` here?